src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -450,112 +449,92 @@ protected virtual string GenerateUniqueId()
             }
         }
 
-        private void RenderUnboundAttribute(string attributeName, Chunk attributeValueChunk)
+        private void RenderUnboundAttribute(TagHelperAttributeTracker attribute)
         {
             // Render children to provide IntelliSense at design time. No need for the execution context logic, it's
             // a runtime feature.
             if (_designTimeMode)
             {
-                if (attributeValueChunk != null)
+                if (attribute.Value != null)
                 {
-                    _bodyVisitor.Accept(attributeValueChunk);
+                    _bodyVisitor.Accept(attribute.Value);
                 }
 
                 return;
             }
 
-            // If we have a minimized attribute there is no value
-            if (attributeValueChunk == null)
-            {
-                _writer
-                    .WriteStartInstanceMethodInvocation(
-                        ExecutionContextVariableName,
-                        _tagHelperContext.ExecutionContextAddMinimizedHtmlAttributeMethodName)
-                    .WriteStringLiteral(attributeName)
-                    .WriteEndMethodInvocation();
-            }
-            else if (attributeValueChunk is PreallocatedTagHelperAttributeChunk)
+            Debug.Assert(attribute.Value != null);
+
+            var attributeValueStyleParameter = $"global::{typeof(HtmlAttributeValueStyle).FullName}.{attribute.ValueStyle}";
+
+            // All simple text attributes will be pre-allocated.
```

Comment:
> Won't minimized attributes also be pre-allocated?

Yup, I'll be more explicit and state minimized instead of grouping the two as simple text.



> Separately, does TagHelperBlockRewriter.ReWrite() ensure preallocated attributes won't have value style NoQuotes?

Yup